stats/stats.go

```diff
@@ -334,6 +334,10 @@ type Sample struct {
 	Value  float64
 }
 
+func (s Sample) Push(ctx context.Context) {
```

Comment:
This also seems bad - too much mixing of concerns. I'd rather not have self-propelled metric samples :sweat_smile: They should be simple data points and something else should push them around.